crates/topos-crypto/src/messages.rs

```diff
@@ -0,0 +1,55 @@
+use ethers::signers::Signer;
+use ethers::signers::{LocalWallet, WalletError};
+use ethers::types::{Address, RecoveryMessage, Signature, SignatureError, H160};
+use ethers::utils::hash_message;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct MessageSigner {
+    pub public_address: Address,
+    wallet: LocalWallet,
+}
+
+impl MessageSigner {
+    pub fn new(private_key: &str) -> Arc<Self> {
+        let wallet: LocalWallet = private_key
+            .parse()
+            .expect("Cannot create wallet for private key");
+        Arc::new(Self {
+            public_address: wallet.address(),
+            wallet,
+        })
+    }
+
+    pub fn sign_message(
+        &self,
+        certificate_id: &[u8],
+        validator_id: &[u8],
+    ) -> Result<Signature, WalletError> {
+        let mut message: Vec<u8> = Vec::with_capacity(certificate_id.len() + validator_id.len());
+        message.extend_from_slice(certificate_id);
+        message.extend_from_slice(validator_id);
```

Comment:
As discuss previously, I would prefer to have a method that sign a payload directly, not matter what's inside.
Having that let us iterate over how we compress or encode our payload without the need of updating this method as well.

At some point, the payload itself will need to be serialized before signing it. It's not up to this method to construct the payload.

This method could just accept any type `T` that implement `Into<Vec<u8>>` at the end.

Reply:
Then the calling method has to construct the payload? What would be the signature like in your mind?

I see it as a first step. Once we need a more general signing message method, we do it?